modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/filter/FileSizeContentDashboardItemFilter.java

```diff
@@ -0,0 +1,253 @@
+	}
+
+	@Override
+	public List<String> getParameterValues() {
+		return Arrays.asList(
+			ParamUtil.getStringValues(_httpServletRequest, getParameterName()));
+	}
+
+	@Override
+	public Type getType() {
+		return Type.SUBMENU;
+	}
+
+	@Override
+	public String getURL() {
+		return null;
+	}
+
+	private DropdownItem _getDropdownItem(SizeType sizeType) {
+		return DropdownItemBuilder.setActive(
+			_isSelected(sizeType)
+		).setHref(
+			_getURL(sizeType)
+		).setLabel(
+			_getLabel(sizeType)
+		).build();
+	}
+
+	private String _getLabel(SizeType sizeType) {
+		StringBundler sb = new StringBundler(7);
+
+		sb.append(_language.get(_httpServletRequest, sizeType.getType()));
+		sb.append(StringPool.SPACE);
+		sb.append(StringPool.OPEN_PARENTHESIS);
+
+		if (sizeType == SizeType.LARGE) {
+			sb.append(StringPool.GREATER_THAN);
+			sb.append(
+				TextFormatter.formatStorageSize(
+					sizeType.getStartValue(), _httpServletRequest.getLocale()));
+		}
+		else if (sizeType == SizeType.MEDIUM) {
+			sb.append(
+				TextFormatter.formatStorageSize(
+					sizeType.getStartValue(), _httpServletRequest.getLocale()));
+			sb.append(StringPool.LESS_THAN_OR_EQUAL);
+			sb.append(
+				TextFormatter.formatStorageSize(
+					sizeType.getEndValue(), _httpServletRequest.getLocale()));
+		}
+		else {
+			sb.append(StringPool.LESS_THAN_OR_EQUAL);
+			sb.append(
+				TextFormatter.formatStorageSize(
+					sizeType.getEndValue(), _httpServletRequest.getLocale()));
+		}
+
+		sb.append(StringPool.CLOSE_PARENTHESIS);
+
+		return sb.toString();
+	}
+
+	private String _getURL(SizeType sizeType) {
+		PortletResponse portletResponse =
+			(PortletResponse)_httpServletRequest.getAttribute(
+				JavaConstants.JAVAX_PORTLET_RESPONSE);
+
+		String url = HttpComponentsUtil.removeParameter(
+			_portal.getCurrentCompleteURL(_httpServletRequest),
+			portletResponse.getNamespace() + getParameterName());
+
+		return HttpComponentsUtil.addParameter(
+			url, portletResponse.getNamespace() + getParameterName(),
+			sizeType.getType());
+	}
+
+	private boolean _isSelected(SizeType sizeType) {
+		List<String> parameterValues = getParameterValues();
+
+		if (ListUtil.isEmpty(parameterValues)) {
+			return false;
+		}
+
+		return Objects.equals(parameterValues.get(0), sizeType.getType());
+	}
+
+	private static final long _FILE_SIZE_MEDIUM = 1024 * 1024;
+
+	private static final long _FILE_SIZE_SMALL = 150 * 1024;
+
+	private final HttpServletRequest _httpServletRequest;
+	private final Language _language;
+	private final Portal _portal;
+
+	private enum SizeType {
+
+		LARGE("large", _FILE_SIZE_MEDIUM + 1, Long.MAX_VALUE),
+		MEDIUM("medium", _FILE_SIZE_SMALL + 1, _FILE_SIZE_MEDIUM),
+		SMALL("small", 0, _FILE_SIZE_SMALL);
+
+		public static SizeType parse(String type) {
+			if (Validator.isBlank(type)) {
+				return null;
+			}
+
+			for (SizeType sizeType : values()) {
+				if (Objects.equals(sizeType.getType(), type)) {
+					return sizeType;
+				}
+			}
+
+			return null;
+		}
```

Comment:
Why not use `valueOf` here? If it is because of the exn it throws, can you use `valueOf` in the implementation of `parse`?